src/llvm_backend.hpp

```diff
@@ -15,6 +15,9 @@
 #include <llvm-c/Object.h>
 #include <llvm-c/BitWriter.h>
 #include <llvm-c/DebugInfo.h>
+#if LLVM_VERSION_MAJOR >= 17 && (LLVM_VERSION_MINOR > 0 || (LLVM_VERSION_MINOR == 0 && LLVM_VERSION_PATCH > 0))
```

Comment:
I think testing for `LLVM_VERSION_MAJOR >= 17` is all that is necessary. Checking for the minor/major distinctions is probably not a good idea.

Reply:
You're right, it should match the `LB_USE_NEW_PASS_SYSTEM` check below.